apps/internal/oauth/ops/authority/authority.go

```diff
@@ -133,6 +133,8 @@ type AuthParams struct {
 	CodeChallengeMethod string
 	// Prompt specifies the user prompt type during interactive auth.
 	Prompt string
+	// IsConfidentialClient specifies if it is a confidential client
+	IsConfidentialClient bool
```

Comment:
is this just a proxy for AuthorizationType == accesstokens.ATConfidential?

Reply:
Yes it is, but we set these only in certain structs and we are extracting the information we need for the cache key mostly from authParams,

I think if we do not want it here in AuthParams, we can add it as a parameter to the CacheKey() function. What do you think ?